helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -182,7 +200,7 @@ private void createLiveInstance() {
     do {
       retry = false;
       try {
-        _zkclient.createEphemeral(liveInstancePath, liveInstance.getRecord());
+        _zkclient.createEphemeral(liveInstancePath, liveInstance.getRecord(), sessionId);
         LOG.info("LiveInstance created, path: " + liveInstancePath + ", sessionId: " + liveInstance.getEphemeralOwner());
       } catch (ZkNodeExistsException e) {
```

Comment:
Why do we catch other exception and throw out HelixException (like what we do in line 264-270?

Reply:
Agreed, rollback is safer.
But consider the possible rollback fails condition, shall we try to create the new live instance node first? (If this is not the current behavior)

Reply:
For ZkSesssionMismatch Exception, we should not throw, we should just cleanup (if there is anything) and then return.

Reply:
If we don't throw exception and just return, createLiveInstance may be treated as success and so carryOverPreviousCurrentState and setupMsgHandler will run. So I would like to follow the original logic to throw an exception if the ephemeral creation fails. The helix exception would be handled by ZkEventThread's run method: log the event's running failure.